R/get_phylopic.R

```diff
@@ -37,10 +37,24 @@ get_phylopic <- function(uuid = NULL, format = "vector") {
   image_info <- phy_GET(file.path("images", uuid))$`_links`
   if (format %in% c("64", "128", "192")) { # get thumbnail url
     thumbs <- image_info$thumbnailFiles
-    url <- thumbs$href[grepl(format, thumbs$sizes)]
+    ind <- grepl(format, thumbs$sizes)
+    if (!any(ind)) {
+      ind <- 1
+      warning(paste("No raster image with dimension", format, "available.",
+                    "Returning raster image with dimensions", rasters$sizes[1],
+                    "instead."))
+    }
+    url <- thumbs$href[ind]
   } else if (format %in% c("512", "1024", "1536")) { # get raster url
     rasters <- image_info$rasterFiles
-    url <- rasters$href[grepl(format, rasters$sizes)]
+    ind <- grepl(format, rasters$sizes)
+    if (!any(ind)) {
+      ind <- 1
+      warning(paste("No raster image with dimension", format, "available.",
```

Comment:
```suggestion
      warning(paste("No raster image with dimensions", format, "available.",
```
Here you use dimension but dimensions in L54. The same in L43 and L44.

Reply:
Ah, so those spellings are/were intentional because the first value in each message is just a single number, hence "dimension", whereas the second number in each message is two values (e.g., "512x300"), hence "dimensions"

Reply:
That makes absolute sense then. Do you think this could be potentially confusing for a user though? After receiving such a warning, they might want to update their script to use the specific dimension to prevent getting warnings? Should they always use the first value? If so, maybe we should just return this?

Reply:
I thought about this, but sometimes the phylopic is landscape, so the larger number (which I believe is always the format value) isn't always the first value in the dimensions (e.g., "300x512"). I suppose we could do some string manipulation to extract the larger of the two numbers?

Reply:
I see! Yes, that's a bit trickier. If the format value is always the larger value then it should be relatively straightforward as you've suggested. I don't think it is a huge issue as it is though... I can just imagine a few puzzled faces every now and then. I'll let you decide!

Reply:
So, I was about to implement this, but then noticed that if the only available size is smaller than the options we currently provide (in the case of c8f71c27-71db-4b34-ac2d-e97fea8762cf, the largest size raster is 210x143), then the user won't be able to supply that format to get_phylopic as it is currently implemented. We can expand the format options to any number (and then pick the closest size, maybe?), but I feel like that opens a whole new can of worms that I think is outside of the scope of this PR.